src/vivarium_public_health/risks/distributions.py

```diff
@@ -141,87 +173,91 @@ def on_initialize_simulants(self, pop_data: SimulantData) -> None:
     # Public methods #
     ##################
 
-    def ppf(self, q):
-        if not q.empty:
-            q = clip(q)
-            weights = self.lookup_tables["ensemble_distribution_weights"](q.index)
+    def ppf(self, quantiles: pd.Series) -> pd.Series:
+        if not quantiles.empty:
+            quantiles = clip(quantiles)
+            weights = self.lookup_tables["ensemble_distribution_weights"](quantiles.index)
             parameters = {
-                name: parameter(q.index) for name, parameter in self.parameters.items()
+                name: param(quantiles.index) for name, param in self.parameters.items()
             }
-            ensemble_propensity = self.population_view.get(q.index).iloc[:, 0]
-            x = EnsembleDistribution(weights, parameters).ppf(q, ensemble_propensity)
+            ensemble_propensity = self.population_view.get(quantiles.index).iloc[:, 0]
+            x = rd.EnsembleDistribution(weights, parameters).ppf(
+                quantiles, ensemble_propensity
+            )
             x[x.isnull()] = 0
         else:
             x = pd.Series([])
         return x
 
 
-class ContinuousDistribution(Component):
+class ContinuousDistribution(RiskExposureDistribution):
     #####################
     # Lifecycle methods #
     #####################
 
-    def __init__(self, risk, mean, sd, distribution=None):
-        super().__init__()
-        self.risk = EntityString(risk)
-        self._distribution = distribution
-        self._parameters = self.get_parameters(mean, sd)
+    def __init__(self, risk: "Risk") -> None:
+        super().__init__(risk)
+        self._distribution = {
+            "normal": rd.Normal,
+            "lognormal": rd.LogNormal,
+        }[risk.distribution_type]
 
-    def setup(self, builder: Builder) -> None:
-        # todo update to have flexible columns for lookup table
-        self.parameters = builder.lookup.build_table(
-            self._parameters, key_columns=["sex"], parameter_columns=["age", "year"]
-        )
+    #################
+    # Setup methods #
+    #################
 
-    ##########################
-    # Initialization methods #
-    ##########################
+    def build_all_lookup_tables(self, builder: "Builder") -> None:
+        value_columns = builder.data.value_columns()(f"{self.risk}.exposure")
+        index = [col for col in self.exposure_data.columns if col not in value_columns]
+
+        mean = self.exposure_data.set_index(index)[value_columns].squeeze(axis=1)
+        sd = self.standard_deviation.set_index(index)[value_columns].squeeze(axis=1)
+        parameters = self._distribution.get_parameters(mean=mean, sd=sd)
+
+        self.lookup_tables["parameters"] = self.build_lookup_table(
+            builder, parameters.reset_index(), list(parameters.columns)
+        )
 
-    def get_parameters(self, mean, sd):
-        index = ["sex", "age_start", "age_end", "year_start", "year_end"]
```

Comment:
Where are all of these columns being stored now?

Reply:
They are deduced from the data used as the source of the lookup table. This is the work that Jim and I have been doing for more flexible lookup tables.